pkg/ocicni/ocicni.go

```diff
@@ -467,33 +467,34 @@ func (plugin *cniNetworkPlugin) GetPodNetworkStatus(podNetwork PodNetwork) ([]cn
 
 	results := make([]cnitypes.Result, 0)
 	if err := plugin.forEachNetwork(&podNetwork, func(network *cniNetwork, ifName string, podNetwork *PodNetwork) error {
-		version := "4"
-		ip, mac, err := getContainerDetails(plugin.nsManager, podNetwork.NetNS, ifName, "-4")
-		if err != nil {
-			ip, mac, err = getContainerDetails(plugin.nsManager, podNetwork.NetNS, ifName, "-6")
-			if err != nil {
-				return err
-			}
-			version = "6"
-		}
+		ips := make([]*cnicurrent.IPConfig, 0)
+		interfaces := make([]*cnicurrent.Interface, 0)
 
-		// Until CNI's GET request lands, construct the Result manually
-		results = append(results, &cnicurrent.Result{
-			CNIVersion: "0.3.1",
-			Interfaces: []*cnicurrent.Interface{
-				{
+		for _, version := range []string{"4", "6"} {
+			ip, mac, err := getContainerDetails(plugin.nsManager, podNetwork.NetNS, ifName, "-"+version)
+			if err == nil {
+				interfaces = append(interfaces, &cnicurrent.Interface{
```

Comment:
in this way we also skip errors that we should probably report back, but I don't see any other immediate way of solving it without requiring `getContainerDetails` to return different  types of errors, so it is probably fine